spec/tasks/metrics_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+
+RSpec.describe "Metrics Overview task", type: :task do
+  before do
+    Rake::Task.clear
+    PracticalDeveloper::Application.load_tasks
+  end
+
+  describe "#overview" do
+    let(:event_name) { "Admin Overview Link Clicked" }
+    let(:click_target) { "https://forem.gitbook.io/forem-admin-guide/quick-start-guide" }
+
+    it "returns the event count and target for admin overview events" do
+      create(:ahoy_event, name: event_name, properties: {
+               action: "click", target: click_target
+             })
+
+      expect { Rake::Task["metrics:overview"].invoke }.to output(
+        "Admin Overview Link Tracking for localhost:3000:\n#{click_target}: 1\n",
+      ).to_stdout
```

Comment:
Fun fact: I initially tried to test this by doing some weird "reassign stdout and try to capture the rake task output to check what it is" nonsense...

```ruby
rake_task_output = StringIO.new
$stdout = rake_task_output # Temporarily assign global stdout to a new StringIO instance

Rake::Task["metrics:overview"].invoke

$stdout = STDOUT # Reset global stdout to STDOUT constant
Rake.application["metrics:overview"].reenable


expect(rake_task_output.string).to include(Ahoy::Event.overview_link_clicks.length.to_s)
expect(rake_task_output.string).to include(click_target)
```
which actually passed!

....and then Rubocop told me in my pre-commit hook to `use to_stdout instead of mutating $stdout` and that's when I realized that there was a Rspec matcher for that 🤦🏽‍♀️ Anyways, I learned some fun stuff about `STDOUT` and `$stdout` in ruby so that was still fun! 😃

Reply:
TIL!

Reply:
Me too. OMG checking the output to STDOUT, a game changer! :D 

Rubocop FTW ahhahaha